modules/internal/CString.chpl

```diff
@@ -382,7 +382,8 @@ module CString {
   */
 
   proc c_string.writeThis(x: Writer) {
-    x.write(this);
+    const tmpStr = new string(this, owned=false, needToCopy=false);
+    x.write(tmpStr);
```

Comment:
Could you say a little bit, perhaps in a comment, if you havn't already elsewhere in the file, if you expect this to work multi-locale? My personal expectation is that c_string.writeThis just can't work if the channel locale != the c string locale. That's because the writeThis is called within a writeln (e.g.) which has an on channel.home in it. So I'm saying that in a multilocale situation, by the time we got to writeThis, we'd already be on the wrong locale. Does that match your expectation?